app/controllers/cases/notes_controller.rb

```diff
@@ -0,0 +1,51 @@
+module Cases
+  class NotesController < ApplicationController
+    before_action :set_case, :set_date_of_birth
+
+    def create
+      begin
+        @case.state_machine.add_note_to_case!(
+          acting_user: current_user,
+          acting_team: case_team,
+          message: params[:case][:message_text]
+        )
+      rescue ActiveRecord::RecordInvalid => err
+        if err.record.errors.include?(:message)
+          flash[:case_errors] = { message_text: err.record.errors[:message] }
+        end
+      ensure
+        redirect_to case_path(@case, anchor: 'case-history')
+      end
+    end
+
+    private
+
+    # need to sort with manager first so that if we are both manager and something else, we don't
+    # try to execute the action with our lower authority (which might fail)
+    def case_team
+      User.sort_teams_by_roles(case_teams).first
+    end
+
+    def case_teams
+      if current_user.teams_for_case(@case).any?
+        current_user.teams_for_case(@case)
+      else
+        current_user.teams
+      end
+    end
+
+    def set_case
+      @case = Case::Base.find(params[:case_id])
+      authorize(@case, :can_add_note_to_case?)
+    end
+
+    # this method is here to fix an issue with the gov_uk_date_fields
+    # where the validation fails since the internal list of instance
+    # variables lacks the date_of_birth field from the json properties
+    #     NoMethodError: undefined method `valid?' for nil:NilClass
+    #     ./app/state_machines/configurable_state_machine/machine.rb:256
+    def set_date_of_birth
+      @case.date_of_birth = @case.date_of_birth
+    end
```

Comment:
This caveat bit of code is now showing up in multiple places. Perhaps time to put a fix into the actual gov_uk gem?

Reply:
The problem with that is that the gem has already gone up a version and we can't use it because it's specific to the new gov uk design system. We could propose a patch to the earlier version, but I wouldn't know where to start with amending the behaviour inside the gem. It's more about the interaction with the json fields so I'm not sure it's a worthwhile use of time anyway

Reply:
Moved the repeated method into a concern